csrc/transform_view.cpp

```diff
@@ -197,24 +203,43 @@ class MergeTransform final : public ViewTransform {
 
     // Assumed to never merge over non-contiguous dimensions.
     IterDomain* outer_id = current_transformed_domain.at(index_);
+    IterDomain* inner_id = current_transformed_domain.at(index_ + 1);
+
+    // When merging, we can only preserve expanded broadcasts if both domains
+    // are broadcasts. Otherwise, we resolve the broadcast and produce an
+    // Iteration output domain.
+    bool realize_expand = !(outer_id->isBroadcast() && inner_id->isBroadcast());
+    auto iter_type = realize_expand ? IterType::Iteration : IterType::Broadcast;
```

Comment:
Move this closer to where it's first used (line 240) or simply inline it (line 240 appears to be the only use of `iter_type`).